src/precice/impl/DataContext.cpp

```diff
@@ -85,20 +87,40 @@ bool DataContext::hasMapping() const
   return hasReadMapping() || hasWriteMapping();
 }
 
-void DataContext::mapData()
+void DataContext::mapData(std::optional<double> after)
 {
+  PRECICE_TRACE(getMeshName(), getDataName(), after.value_or(-666));
   PRECICE_ASSERT(hasMapping());
-  // Execute the mapping
+  PRECICE_ASSERT(after.value_or(0) >= 0.0, "Invalid time");
```

Comment:
Does this assertion really help? We are not dealing with relative times anymore and we do not really know a lot about the valid times to expect (before it was $t\in[0,1]$). Wouldn't it make more sense to just use `unsigned double` here? (btw: I think there is also some potential for refactoring in `BaseCouplingScheme` etc. to use `unsigned double` for all the members tracking absolute time)

Reply:
The trouble with `unsigned double` is that one needs to implement this in the communication backends too.

But you are right. The assertion is not really necessary.

Reply:
Removed it